src/app/features/public-form/public-form.component.ts

```diff
@@ -93,10 +98,24 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
     this.formConfig = publicForms.find(
       (form: PublicFormConfig) => form.route === id || form.getId(true) === id,
     );
+    if (!this.formConfig) {
+      this.publicFormNotFound = {
+        title: $localize`Public Form Not Found`,
+        error: $localize`The form you are looking for is either unavailable or doesn't exist. Please check the link and try again.`,
+      };
+      return;
+    }
 
     this.entityType = this.entities.get(
       this.formConfig.entity,
     ) as EntityConstructor<E>;
+    if (this.ability.cannot("create", this.entityType)) {
+      this.publicFormNotFound = {
+        title: $localize`Access Denied to Public Form - ${this.formConfig.title}`,
+        error: $localize`Unfortunately, access has been denied due to insufficient permissions to access this Public Form. Please contact your system administrator.`,
```

Comment:
```suggestion
        title: $localize`No access to public form "${this.formConfig.title}"`,
        error: $localize`Unfortunately, you do not have the required permissions to submit data through this form. Please contact the person who shared this link with you. In case you have a user account, please make sure you are logged in.`,
```